File: csc209/c2changk/209Labs/Lab1/lab01stuff/getinfo.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAX 256

int main() {
	char stu_name[MAX];
	char classes[MAX];

	/* Note that printf("Please enter your name:\n"); is the same as the 
	line below.  The fprintf call makes it explicit where the output is
	going. */

	fprintf(stdout, "Please enter your name:\n");

	/* fgets is reading a line of input from stdin, and saving it in the 
	 * character array stu_name. We want to record that there is an error
	 * if it wasn't able to read from stdin or if the input was empty (only
	 * an newline character).
	 */
	if((fgets(stu_name, MAX, stdin) == NULL) || (strlen(stu_name) <= 1)) {
		fprintf(stderr, "Error: name was not supplied\n");
		return(1);
	}

	/* Remove the newline character */
	stu_name[strlen(stu_name) - 1] = '\0';

	printf("Please enter your classes this term:\n");
	if(fgets(classes, MAX, stdin) == NULL || strlen(classes) <= 1) {
		fprintf(stderr, "Error: no classes were entered.\n");
		return(1);
	}

	classes[strlen(classes) - 1] = '\0';

	/* Look for the end of the first word in the array, and put a null
	 * termination character in, effectively truncating the list to be
	 * only the first word. 
	 */
 	char *space = index(classes, ' ');
	if (space != NULL) { 
		*space = '\0';
	}

	printf("Hi %s, are you enjoying %s?\n", stu_name, classes);

	return 0;
}

File: csc209/c2changk/209Labs/Lab1/lab01stuff/getinfo.py
#!/usr/local/bin/python3

import sys

# By default, print() writes to standard output

print("Please enter your name:")

# By default input() reads from standard input
stu_name = input()

print("Please enter your classes this term:")
classes = input()

if len(stu_name.strip()) == 0:
	# If an error occurs, we want to write to standard error
	print("Error: name cannot be empty", file=sys.stderr)

	# In this case we choose to terminate the program with a non-zero exit
	# value indicating that an error occurred.
	exit(1)
	
if len(classes.strip()) == 0:
	print("Error: you are at least taking one class", file=sys.stderr)
	exit(1)
else:
	data = classes.split(" ")
	

print("Hi ", stu_name, ", are you enjoying ", data[0], "?", sep="")
